### PR TITLE
[16.0][MIG] coupon_incompatibility: Migration to version 16.0

### PR DESCRIPTION
**Pull request by ernesto-garcia-tecnativa**
@tecnativa TT44318

### CODE REVIEW

**Comment by ernesto-garcia-tecnativa** [comment]
@pedrobaeza @chienandalu please review

**Comment by pedrobaeza** [comment]
It should be renamed (with all the history) to `loyalty_incompatibility`.

**Comment by ernesto-garcia-tecnativa** [comment]
> It should be renamed (with all the history) to `loyalty_incompatibility`.

Done and pushed @pedrobaeza

**Comment by pedrobaeza** [comment]
/ocabot migration coupon_incompatibility

Change the message of the last commit to `[MIG] coupon_incompatibility: Migration to 16.0 and renamed to loyal_incompatibility`.

**Comment by pedrobaeza** [comment]
Rename also the POT and replace the PO/POT content with the proper module name.

File: loyalty_incompatibility/views/sale_coupon_program_views.xml
Comment (chienandalu) [inline loyalty_incompatibility/views/sale_coupon_program_views.xml:1]: Rename to loyalty_program_views

File: loyalty_incompatibility/views/sale_coupon_program_views.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="sale_loyalty_program_view_form_common" model="ir.ui.view">
```
Comment (chienandalu) [inline loyalty_incompatibility/views/sale_coupon_program_views.xml]:
```suggestion
    <record id="loyalty_program_view_form" model="ir.ui.view">
```

File: loyalty_incompatibility/models/coupon_program.py
Comment (chienandalu) [inline loyalty_incompatibility/models/coupon_program.py:1]: rename to loyalty_program.py

File: loyalty_incompatibility/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from . import coupon_program
```
Comment (chienandalu) [inline loyalty_incompatibility/models/__init__.py]:
```suggestion
from . import loyalty_program
```

File: loyalty_incompatibility/tests/test_coupon_incompatibility.py
Comment (chienandalu) [inline loyalty_incompatibility/tests/test_coupon_incompatibility.py:1]: Rename to test_loyalty_incompatibility.py

File: loyalty_incompatibility/tests/test_coupon_incompatibility.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Copyright 2021 Tecnativa - David Vidal
+# Copyright 2023 Tecnativa - Stefan Ungureanu
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo.tests import TransactionCase
+
+
+class CouponIncompatibilityCase(TransactionCase):
```
Comment (chienandalu) [inline loyalty_incompatibility/tests/test_coupon_incompatibility.py]:
```suggestion
class LoyaltyIncompatibilityCase(TransactionCase):
```

File: loyalty_incompatibility/models/coupon_program.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+class LoyaltyProgram(models.Model):
+    _inherit = "loyalty.program"
+
+    incompatible_promotion_ids = fields.Many2many(
+        comodel_name="loyalty.program",
+        relation="sale_coupon_program_incompatibility_rel",
+        column1="program_id",
+        column2="incompatible_program_id",
+        inverse="_inverse_incompatible_promotion_ids",
+        string="Incompatible Promotions",
```
Comment (chienandalu) [inline loyalty_incompatibility/models/coupon_program.py]:
```suggestion
        string="Incompatible Promotions",
        domain="[('id', '!=', self.id)]",
```

**Review by chienandalu — CHANGES_REQUESTED** [review]
Thanks @ernesto-garcia-tecnativa 

- Keep it open so we can test it in your sale_loyalty_incompatibility migration
- Some remarks:

**Comment by ernesto-garcia-tecnativa** [comment]
@chienandalu please review

**Comment by chienandalu** [comment]
> @chienandalu please review

Please do the sale_coupon_incompatibility migration adding this PR as a dependency. Here's the guide to do so: https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference(s)-to-another-pull-request(s)

**Pull request closed without merging**